STYLEGUIDE.md

```diff
@@ -72,3 +76,74 @@ import 'camera.dart';
 import 'game.dart';
 import 'viewport.dart';
 ```
+
+### Identifiers
+
+Use all [effective dart patterns](https://dart.dev/guides/language/effective-dart)
+
+* Avoid using global constants, but if you do use, name it prefixing with a `k`.
+
+```dart
+// Do not:
+const someConstant = 42;
+
+// Do instead:
+const kSomeConstant = 42;
+```
+
+### Comments
+
+* Add an identifier for TODO comments
+* Use fully capitalized TODO notation
+
+```dart
+
+// Do not:
+// Todo: This thing should be that thing
+final thisThing = 13;
+
+// Do instead:
+// TODO(wolfenrain): This thing should be that thing
+const thisThing = 13;
+```
+
+### Asserts
+
+* Use asserts to detect contract violation.
+* Use asserts to run debug only code.
+
+Example:
+````dart
+void something(int smaller, int bigger) {
+  assert(small < bigger, "smaller is not smaller than bigger");
+  assert((){
+    // debug only code
+  });
+  
+  //...
+}
+
+````
+
+
+
+## Code documentation 
+
+Specific guides for dartdoc documentation.
+
+#### Use impersonal tone
+
+Avoid using words such as "you", "ours" and "yours" on code documentation
+
+#### Consider adding examples
+
+Some elements may benefit from a simple usage example.
+
+#### Avoid useless code documentation
+
+Avoid documentation that just repeats the obvious. 
+For example, `void doStuff()` be documented as "Method that does stuff".
+
+#### Consider adding linkage between docs
+
+Consider adding a "See also" section to element documentation
```

Comment:
also make sure to mark code elements with `[]`

Reply:
Well, this is basic dartdoc, do we really include this here?

Reply:
people might not know about it, I don't see why not make our preferred styles explicit